pkg/apis/topology/v1alpha1/types_partitionset.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// PartitionSet defines a target domain and dimensions to slice and dice it for the creation of partitions of shards.
+//
+// +crd
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster,categories=kcp
+// +kubebuilder:printcolumn:name="Count",type="string",JSONPath=".status.count",description="Count of the partitions belonging the PartitionSet"
+// +kubebuilder:printcolumn:name="Age",type="date",JSONPath=".metadata.creationTimestamp"
+type PartitionSet struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec holds the desired state.
+	//
+	// +required
+	// +kubebuilder:validation:Required
+	Spec PartitionSetSpec `json:"spec,omitempty"`
+
+	// Status holds information about the current status
+	//
+	// +optional
+	Status PartitionSetStatus `json:"status,omitempty"`
+}
+
+// PartitioniSetSpec records dimensions and a target domain for the partitioning.
+type PartitionSetSpec struct {
+	// dimensions
+	//
+	// +optional
+	Dimensions []string `json:"dimensions,omitempty"`
+
+	// selector is a label selector that filters shard targets.
+	//
+	// +optional
+	Selector *metav1.LabelSelector `json:"selector,omitempty"`
+}
+
+// PartitioniSetStaus records the status of the PartitionSet.
+type PartitionSetStatus struct {
+	// Count is the total number of partitions.
+	//
+	Count int16 `json:"count,omitempty"`
```

Comment:
If we're worried about size, we shouldn't make this signed :)

Reply:
ack